spec/plugins/data_table/server_side.js

```diff
@@ -290,15 +299,43 @@ describe("DataTable Plugin", function() {
 
     it("should default to GET when an AJAX method is not specified", function() {
       app.view.dataTable("AjaxMethodTestTable", {
-        rowClassName : "R",
-        serverSide : true,
+        rowClassName: "R",
+        serverSide: true
       });
 
-      table = new app.Views.AjaxMethodTestTable({ collection : collection });
+      table = new app.Views.AjaxMethodTestTable({collection: collection});
       table.render();
 
       expect(jasmine.Ajax.requests.mostRecent().method).toEqual("GET");
     });
+
+    it("should pass paging and sort in ajax params", function() {
+      table = new app.Views.T({ collection : collection });
+      table.render();
+
+      table.serverParams({ monkey : "chicken" });
+      expect(jasmine.Ajax.requests.mostRecent().url).toMatch("iSortCol_0=1");
+      expect(jasmine.Ajax.requests.mostRecent().url).toMatch("sSortDir_0=desc");
+      expect(jasmine.Ajax.requests.mostRecent().url).toMatch("sEcho=2");
+
+      expect(jasmine.Ajax.requests.mostRecent().url).not.toMatch("sort_by");
+      expect(jasmine.Ajax.requests.mostRecent().url).not.toMatch("sort_dir");
+      expect(jasmine.Ajax.requests.mostRecent().url).not.toMatch("request_id");
+    });
+
+    it("should pass simpler paging and sort param names when in simple_params mode", function() {
```

Comment:
Added test cases for no sort at all, and no sort name but passing a sort dir.
Also fixed a bug around sorting on first column (index 0, so the `if (0)` didn't work so well..)